### ISSUE TITLE
Flaw in sample timer initialization (interval type)

### ISSUE THREAD

**Issue by ricmua** [ricmua/ros_force_dimension#5]
There is a fairly basic flaw in the `on_activate` method of the node. An `int` type is used for `sample_interval_ms`, instead of float, while [setting the sample timer]. This effectively sets an upper limit of 1000 Hz on the sample frequency. This won't be enough for smooth robot control.

[setting the sample timer]: https://github.com/ricmua/ros_force_dimension/blob/0ff18426c4acf08fb5be5d397024a74defe3f3fa/src/node.cpp#L155

**Comment by ricmua** [comment]
The Force Dimension node was patched and the sampling rate tested again. The initialization status messages report the correct sampling interval (2000 Hz):

```bash
path/to/workspace$ ros2 run force_dimension node --ros-args -r __node:=manipulandum --params-file config/novint_falcon.yaml -p disable_hardware:=true
[INFO] [1687381829.027167359] [robot.manipulandum]: Initializing the Force Dimension interface.
[INFO] [1687381829.027257727] [robot.manipulandum]: Force Dimension device detected: hardware disabled
[INFO] [1687381829.027272342] [robot.manipulandum]: Force Dimension interface initialized.
[INFO] [1687381829.027283315] [robot.manipulandum]: Communication refresh rate: -1 kHz
[INFO] [1687381829.027311650] [robot.manipulandum]: Sample timer initialized: Interval (ms) = 0.500000
[INFO] [1687381829.027323648] [robot.manipulandum]: BASELINE EFFECTOR MASS: -1.000000 kg
[INFO] [1687381829.027335466] [robot.manipulandum]: Effector mass set to -1.000000 kg.
[INFO] [1687381829.027349239] [robot.manipulandum]: Gravity compensation: Enabled
```

And the `hz` command confirms:

```bash
path/to/workspace$ ros2 topic hz /robot/feedback/position
average rate: 2000.256
	min: 0.000s max: 0.001s std dev: 0.00003s window: 2001
average rate: 2000.129
	min: 0.000s max: 0.001s std dev: 0.00003s window: 4002
average rate: 2000.081
	min: 0.000s max: 0.002s std dev: 0.00004s window: 6002
average rate: 1999.930
	min: 0.000s max: 0.002s std dev: 0.00004s window: 8002
```

**Comment by ricmua** [comment]
Resolved with [4843da5].

[4843da5]: https://github.com/ricmua/ros_force_dimension/commit/4843da55576fecdb60e3126e9fd93c49f033e0d5

**Comment by ricmua** [comment]
Useful links:

* https://cplusplus.com/reference/chrono/nanoseconds/
* https://en.cppreference.com/w/cpp/chrono/duration
